Kernel/Library/StdLib.h

```diff
@@ -16,6 +17,25 @@
 #include <stddef.h>
 
 ErrorOr<NonnullOwnPtr<Kernel::KString>> try_copy_kstring_from_user(Userspace<char const*>, size_t);
+
+ErrorOr<void> try_copy_string_from_user(Userspace<char const*>, char*, size_t);
+
+template<size_t Size>
+ErrorOr<void> try_copy_string_from_user_into_static_char_buffer(Userspace<char const*> user_str, FixedCharBuffer<Size>& buffer, size_t user_str_size)
+{
+    if (user_str_size > buffer.fixed_length())
+        return E2BIG;
+    return try_copy_string_from_user(user_str, buffer.unsafe_char_ptr(), user_str_size);
+}
+
+template<size_t Size>
+ErrorOr<void> try_copy_name_from_user_into_static_char_buffer(Userspace<char const*> user_str, FixedCharBuffer<Size>& buffer, size_t user_str_size)
+{
+    if (user_str_size > buffer.fixed_length())
```

Comment:
The fact that the checks for the size of the Kernel's buffer against the untrusted userspace-provided size are in the template function here means that the underlying `try_copy_string_from_user` function is inherently unsafe.

Any other kernel code that calls that function must know that it's their responsibility to check the size beforehand, which seems an unnecessary footgun.